### PR TITLE
Don't run EmulatedBS2_Wii with VolumeDirectory

### PR DESCRIPTION
**Pull request by JosJuice**
VolumeDirectory doesn't support necessities like TMDs, so thanks to 5.0-2172 (18968ab), EmulatedBS2_Wii crashes when the inserted disc is a VolumeDirectory. This PR fixes that.

This PR makes our DOL booting code very similar to our ELF booting code. One exception is that the DOL booting code still always calls SetupBAT. (Note that EmulatedBS2_GC calls SetupBAT even if no disc is inserted.) I'm not sure if there's a point to the difference, but I thought I'd better avoid changing it so that I don't break anything.

### CODE REVIEW

**Comment by JosJuice** [comment]
To be more specific, there was a regression starting with 5.0-2127 that made Dolphin immediately crash when booting any Wii game using VolumeDirectory+DOL. This PR fixes that.

**Comment by Sarkie** [comment]
Interested to see if this is the cause of the failure at booting .dols on Android.

File: Source/Core/Core/Boot/Boot.cpp [hunk 1]
```diff
@@ -351,39 +351,44 @@ bool CBoot::BootUp()
...
+    PowerPC::ppcState.spr[SPR_DBAT0U] = 0x80001fff;
+    PowerPC::ppcState.spr[SPR_DBAT0L] = 0x00000002;
+    PowerPC::ppcState.spr[SPR_DBAT1U] = 0xc0001fff;
+    PowerPC::ppcState.spr[SPR_DBAT1L] = 0x0000002a;
+    PowerPC::ppcState.spr[SPR_DBAT4U] = 0x90001fff;
+    PowerPC::ppcState.spr[SPR_DBAT4L] = 0x10000002;
+    PowerPC::ppcState.spr[SPR_DBAT5U] = 0xd0001fff;
+    PowerPC::ppcState.spr[SPR_DBAT5L] = 0x1000002a;
+    if (dolWii)
+      HID4.SBE = 1;
```
Comment (JosJuice) [inline Source/Core/Core/Boot/Boot.cpp]: I can do that, but what would an appropriate name for the function be? I don't really know what the purpose of the code is, so nothing comes to mind to me.

Comment (BhaaLseN) [inline Source/Core/Core/Boot/Boot.cpp]: Good question. `SetupSPR_Wii`, `SetupBAT_Wii`, `SetupPPCState_Wii` or something along those lines? `_Wii` suffix since GC does something similar; except not everything due to the Wii having more SPRs available.

Comment (JosJuice) [inline Source/Core/Core/Boot/Boot.cpp]: Done. I made it have an `is_wii` parameter so that it can be used in `EmulatedBS2_GC` too.

Comment (JosJuice) [inline Source/Core/Core/Boot/Boot.cpp]: Also done in PR #5406.

File: Source/Core/Core/Boot/Boot.cpp [hunk 2]
```diff
@@ -351,28 +351,9 @@ bool CBoot::BootUp()
...
-    PowerPC::ppcState.spr[SPR_DBAT1L] = 0x0000002a;
-    PowerPC::ppcState.spr[SPR_DBAT4U] = 0x90001fff;
-    PowerPC::ppcState.spr[SPR_DBAT4L] = 0x10000002;
-    PowerPC::ppcState.spr[SPR_DBAT5U] = 0xd0001fff;
-    PowerPC::ppcState.spr[SPR_DBAT5L] = 0x1000002a;
     if (dolWii)
       HID4.SBE = 1;
-    PowerPC::DBATUpdated();
-    PowerPC::IBATUpdated();
+    SetupBAT(/*is_wii*/ true);
```
Comment (BhaaLseN) [inline Source/Core/Core/Boot/Boot.cpp]: Hm, should we call this with `dolWii` (or perhaps `_StartupPara.bWii`) instead?
It probably doesn't make a difference in practise because the GC doesn't have `SPR_{I,D}BAT{4,5}{U,L}` and thus shouldn't look at them, but for accuracy, don't initialize what we don't need?
I dunno tbh, just something to look at.

Comment (JosJuice) [inline Source/Core/Core/Boot/Boot.cpp]: Probably, but I would prefer to do that in a separate PR so that we don't get too many behavior changes in one.

Comment (BhaaLseN) [inline Source/Core/Core/Boot/Boot.cpp]: Sure, but the more you look at something the more things you notice you'd want to change. LGTM then.
